samples/client/petstore/php/psr-18/lib/Api/DefaultApi.php

```diff
@@ -464,4 +464,9 @@ private function createUri(
 
         return $uri;
     }
+
+    private function bodyShouldBeEncoded(string $contentType): bool 
+    {        
+        return preg_match('~^application/(json|[\w!#$&.+-^_]+\+json)\s*(;|$)~', $contentType) === 1;
+    }
```

Comment:
what about adding this function in the api client class instead?

can we name this method as isJsonMime?

minor suggestion: add a docstring explaining what this function does

Reply:
I am not sure, what you mean by "what about adding this function in the api client class instead?"
As this is a method used throughout the code and if not working properly results in Exceptions being thrown, I think it is quite important.
What I could think of, is implementing a separate Class "JsonDetector" or anything with a single method, which could be injected via consturctor DI. Would this be a good solution or are you thinking of anything else?

Reply:
sorry for the confusion, api client class (ApiClient.php) is in client generated by php-dt generator but not php client generator.

shall we put the function in HeaderSelector class instead which has something similar already?

https://github.com/OpenAPITools/openapi-generator/blob/11a483ef92eff347b826d18398191d97406a2d68/samples/client/petstore/php/psr-18/lib/HeaderSelector.php#L88

Reply:
Did you have something like this in mind?